src/System.Private.TypeLoader/src/Internal/Reflection/Execution/AssemblyBinderImplementation.Ecma.cs

```diff
@@ -186,11 +186,11 @@ public PEInfo(RuntimeAssemblyName name, MetadataReader reader, PEReader pe)
                     }
 
                     // Cache missed lookups
-                    if (!foundMatch)
-                    {
-                        PEInfo peinfo = new PEInfo(refName, null, null);
-                        s_ecmaLoadedAssemblies.Add(peinfo);
-                    }
+                    // if (!foundMatch)
```

Comment:
Should this code be put back?

Reply:
The interpreter is gonna fail if this is still in there. Can't really say whether the code is essential for other scenarios which is why I just left it commented out for now

Reply:
Someone will need to understand the code. I didn't have a chance to look into it deeply yet. Maybe @davidwrighton could help us here? This code was added as part of support for satellite assemblies, but breaks the `Assembly.Load(byte[])` scenario.

Reply:
I'll admit that I don't know what satellite assemblies are. What scenarios are they typically used?

Reply:
https://blogs.msdn.microsoft.com/global_developer/2011/07/22/introduction-to-satellite-assemblies/ - you can put managed resourced for a specific locale in them. I'm not sure if this actually works with the CoreRT compiler, but it's a supported scenario for .NET Native for UWP apps (this code is shared with the closed source runtime/compiler).

Reply:
Sadly, this is necessary for satellite assembly work. What you'll need to do is add a Boolean flag to the function to indicate whether or not the caching should happen, and then thread it through such that for byte[] loads, the failure isn't cached when we do the speculative lookup to see if the assembly has already been loaded.